etc/run-clang-tidy-cached.cc

```diff
@@ -125,7 +125,8 @@ static constexpr ConfigValues kConfig = {
     .start_dir = "src/",
     .file_exclude_re
     = "src/sta/"  // Don't check 3rd-party submodule
-      "|codeGenerator/templates/",
+      "|codeGenerator/templates/"
+      "|build/", // Don't check generated code
```

Comment:
@hzeller FYI

Reply:
This should not be necessary, as our `.start_dir = "src"`, meaning we don't follow directories and files outside that to run `clang-tidy` on.

If there are reports in the clang-tidy output that do point to `build/`, then it is probably because clang-tidy followed some generated include somewhere and reports things there (do you have an example ?). It won't be fixed by this patch though.

I actually have a fix for that by filtering out messages that don't refer to the file in question, I can probably send a PR later this evening.

Reply:
I did see reports in the swig generated code which isn't included anywhere afaik.  I suspect it is because the cmake in the src references it.